test/extensions/tracers/skywalking/skywalking_client_config_test.cc

```diff
@@ -0,0 +1,90 @@
+#include "extensions/tracers/skywalking/skywalking_client_config.h"
+
+#include "test/mocks/common.h"
+#include "test/mocks/server/tracer_factory_context.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace testing;
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace SkyWalking {
+
+class SkyWalkingClientConfigTest : public testing::Test {
+public:
+  void setupSkyWalkingClientConfig(const std::string& yaml_string) {
+    auto& local_info = context_.server_factory_context_.local_info_;
+
+    ON_CALL(local_info, clusterName()).WillByDefault(ReturnRef(test_string));
+    ON_CALL(local_info, nodeName()).WillByDefault(ReturnRef(test_string));
+
+    envoy::config::trace::v3::SkyWalkingConfig proto_config;
+    TestUtility::loadFromYaml(yaml_string, proto_config);
+
+    client_config_ =
+        std::make_unique<SkyWalkingClientConfig>(context_, proto_config.client_config());
+  }
+
+protected:
+  NiceMock<Envoy::Server::Configuration::MockTracerFactoryContext> context_;
+
+  std::string test_string = "ABCDEFGHIJKLMN";
+
+  SkyWalkingClientConfigPtr client_config_;
+};
+
+static const std::string SKYWALKING_CONFIG_WITH_CLIENT_CONFIG = R"EOF(
```

Comment:
no global static std::string is allowed, this is source of test flakes. just do it inline or `constexpr std::string_view`

Reply:
OK.